### PR TITLE
Updating namespace.py to solve issue #801

### LINKED ISSUE #801: Problem with prefixes created for URIs containing %20
The result of runnig this code
```python
from rdflib import Namespace, Graph, BNode, Literal

graph = Graph()
namespace = Namespace('http://example.org/')
graph.bind('', namespace)
node = BNode()

graph.add((node, namespace['first%20name'], Literal('John')))
print(graph.serialize(format='turtle').decode())
```
is
```
@prefix : <http://example.org/> .
@prefix ns1: <http://example.org/first%20> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

[] ns1:name "John" .
```
I expect that triple would be displayed as 
```
[] :first%20name "John" .
```

### PR DESCRIPTION
**Pull request by aayush17002**
Added acceptance clause for "%" in Allowed name chars.

Closes #801

### CODE REVIEW

**Comment by white-gecko** [comment]
Were you able to check if this also results in valid results for the other serialization formats? https://github.com/RDFLib/rdflib/issues/801#issuecomment-353026657

**Comment by aayush17002** [comment]
Yes on making the changes described in my code I was able to get valid results other serialization formats also.

**Comment by aayush17002** [comment]
Sir,
Are there any other changes required, If not, please accept my solution.

File: test/test_issue801 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+  """
+Issue 715 - path query chaining issue
+Some incorrect matches were found when using oneOrMore ('+') and
+zeroOrMore ('*') property paths and specifying neither the
+subject or the object.
```
Comment (darrengarvey) [inline test/test_issue801:5]: This comment looks out of date.

File: test/test_issue801 [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class TestIssue801(unittest.TestCase):
+
+    def test_issue_801(self):
+        g = Graph()
+        example = Namespace('http://example.org/')
+        g.bind('', example)
+        node = BNode()
+        g.add((node, example['first%20name'], Literal('John')))
+        print(g.serialize(format="turtle").decode())
```
Comment (darrengarvey) [inline test/test_issue801:19]: Shouldn't this be checking the result of the serialisation?

File: test/test_issue801 (hunk 1 above)
Comment (aayush17002) [reply test/test_issue801]: @darrengarvey I have updated the comment

File: test/test_issue801 (hunk 2 above)
Comment (aayush17002) [reply test/test_issue801]: @darrengarvey I have added the assert statement for checking the result

File: test/test_issue801.py [hunk 3]
```diff
@@ -13,7 +13,8 @@ def test_issue_801(self):
         g.bind('', example)
         node = BNode()
         g.add((node, example['first%20name'], Literal('John')))
-        print(g.serialize(format="turtle").decode())
+        self.assertEqual(g.serialize(format="turtle").decode().split("\n")[-3] == '[] :first%20name "John" .', True)
```
Comment (darrengarvey) [inline test/test_issue801.py:16]: Can this instead be:

```
self.assertEqual(g.serialize(format="turtle").decode().split("\n")[-3],
                '[] :first%20name "John" .')
```

That way if the test fails you get to see what both sides of the equality comparison are. Plus keeping the line length shorter and making the string easier to see.

File: test/test_issue801.py [hunk 4]
```diff
@@ -13,7 +13,8 @@ def test_issue_801(self):
...
+        print (g.serialize(format="turtle").decode())
```
Comment (darrengarvey) [inline test/test_issue801.py:17]: This is going to make test run output noisy. I don't see other tests in rdflib printing output, so do you need this?

File: test/test_issue801.py (hunk 3 above)
Comment (aayush17002) [reply test/test_issue801.py]: @darrengarvey I have applied the requested changes

File: test/test_issue801.py (hunk 4 above)
Comment (aayush17002) [reply test/test_issue801.py]: @darrengarvey I have applied the requested changes

**Comment by nicholascar** [comment]
@aayush17002 please merge in `master` branch since a whole host of changes were recently merged in to remove Python 2 code. This will trigger a re-checking of the code by Travis. Then it may be ready for merging.

**Comment by aayush17002** [comment]
@nicholascar I have merged into the master branch again as suggested by you

**Pull request merged**